### PR TITLE
Refactor to reflect current product names

### LINKED ISSUE #292: Renaming Contracts to Reflect Current Product Names
TODO: need to make a proper checklist but this is a placeholder issue that I will fill out at a later date.

We updated the branding on several products, including Bonding -> Staking; Ubiquity Auto Redeem -> Ubiquity Credit; and more. Currently they are only reflected in the UI but we should update the contracts to reflect the updated names.

***

        I understand your concern about renaming. If we create a new issue for renaming and resolve them on it, we will never miss any renaming stuff. we can also have checklist on the issue which gives clear steps to do.

_Originally posted by @0xcodercrane in https://github.com/ubiquity/ubiquity-dollar/pull/261#discussion_r997389498_

### PR DESCRIPTION
**Pull request by ryzhak**
PR for https://github.com/ubiquity/ubiquity-dollar/issues/292
 
 Notice:
 1. There is no refactoring in the `ubiquistick` folder because it wasn't mentioned in the task
 2. renamed `CurveUADIncentive.sol` to `transfer-hooks/CurveDollarIncentive.sol` instead of `transfer-hooks/dollar3pool.sol` because IMHO filename should be equal to contract name
 3. Frontend should be completely broken after these changes (need a task to fix it)
 
 When this PR is merged I will update this wiki page: https://github.com/ubiquity/ubiquity-dollar/wiki/18.-Smart-Contracts

### CODE REVIEW

**Comment by ryzhak** [comment]
FYI: somehow I don't have rights to add reviewers to this PR

**Comment by 0x4007** [comment]
> FYI: somehow I don't have rights to add reviewers to this PR

If you follow the PR template (uses keyword "resolves") it will automatically associate this PR with the bounty/issue. This is important for fastest response/organization of the issues/bounties. No need to assign reviewers as it generally is @0xcodercrane and myself. We can assign amongst ourselves!

File: packages/contracts/src/dollar/UbiquityChef.sol [hunk 1]
```diff
@@ -53,14 +52,14 @@ contract MasterChefV2 is ReentrancyGuard {
     // Info of each pool.
     PoolInfo public pool;
     // Info of each user that stakes LP tokens.
-    mapping(uint256 => BondingShareInfo) private _bsInfo;
+    mapping(uint256 => StakingShareInfo) private _ssInfo;
```
Comment (0x4007) [inline packages/contracts/src/dollar/UbiquityChef.sol:55]: Nice attention to detail

File: packages/contracts/src/dollar/core/CreditNFT.sol [hunk 2]
```diff
@@ -1,14 +1,14 @@
...
 pragma solidity ^0.8.3;
 
-import "./ERC1155Ubiquity.sol";
 import "solidity-linked-list/contracts/StructuredLinkedList.sol";
-import "./UbiquityAlgorithmicDollarManager.sol";
+import "../ERC1155Ubiquity.sol";
+import "./UbiquityDollarManager.sol";
 
-/// @title A coupon redeemable for dollars with an expiry block number
+/// @title A Credit NFT redeemable for dollars with an expiry block number
```
Comment (0x4007) [inline packages/contracts/src/dollar/core/CreditNFT.sol:8]: 👍

File: packages/contracts/src/dollar/core/DollarMintCalculator.sol [hunk 3]
```diff
@@ -1,30 +1,30 @@
...
-import "./libs/ABDKMathQuad.sol";
+import "../interfaces/IDollarMintCalculator.sol";
+import "../libs/ABDKMathQuad.sol";
+import "./TWAPOracleDollar3pool.sol";
+import "./UbiquityDollarManager.sol";
 
-/// @title A mock coupon calculator that always returns a constant
-contract DollarMintingCalculator is IDollarMintingCalculator {
+/// @title Calculates amount of dollars ready to be minted when twapPrice > 1
+contract DollarMintCalculator is IDollarMintCalculator {
```
Comment (0x4007) [inline packages/contracts/src/dollar/core/DollarMintCalculator.sol:11]: Using the same tense of the verbs in the names?

File: packages/contracts/src/dollar/core/DollarMintExcess.sol [hunk 4]
```diff
@@ -37,25 +37,25 @@ contract ExcessDollarsDistributor is IExcessDollarsDistributor {
...
+            // curve UbiquityDollar-3CRV liquidity pool
             uint256 tenPercent =
                 excessDollars.fromUInt().div(uint256(10).fromUInt()).toUInt();
             uint256 fiftyPercent =
                 excessDollars.fromUInt().div(uint256(2).fromUInt()).toUInt();
             IERC20Ubiquity(manager.dollarTokenAddress()).safeTransfer(
                 treasuryAddress, fiftyPercent
             );
-            // convert uAD to uGOV-UAD LP on sushi and burn them
+            // convert Ubiquity Dollar to GovernanceToken-UbiquityDollar LP on sushi and burn them
```
Comment (0x4007) [inline packages/contracts/src/dollar/core/DollarMintExcess.sol]:
```suggestion
            // convert Ubiquity Dollar to GovernanceToken-DollarToken LP on sushi and burn them
```

File: packages/contracts/src/dollar/core/DollarMintExcess.sol [hunk 5]
```diff
@@ -81,21 +81,21 @@ contract ExcessDollarsDistributor is IExcessDollarsDistributor {
...
         );
-        uint256 amountUGOV = _swapDollarsForGovernance(amountUAD);
+        uint256 amountGovTokens = _swapDollarsForGovernance(amountDollars);
 
         IERC20Ubiquity(manager.governanceTokenAddress()).safeApprove(
             address(_router), 0
         );
         IERC20Ubiquity(manager.governanceTokenAddress()).safeApprove(
-            address(_router), amountUGOV
+            address(_router), amountGovTokens
```
Comment (0x4007) [inline packages/contracts/src/dollar/core/DollarMintExcess.sol]:
```suggestion
            address(_router), amountGovernanceTokens
```

File: packages/contracts/src/dollar/core/DollarMintExcess.sol [hunk 6]
```diff
@@ -81,21 +81,21 @@ contract ExcessDollarsDistributor is IExcessDollarsDistributor {
...
         );
 
         // deposit liquidity and transfer to zero address (burn)
         _router.addLiquidity(
             manager.dollarTokenAddress(),
             manager.governanceTokenAddress(),
-            amountUAD.toUInt(),
-            amountUGOV,
+            amountDollars.toUInt(),
+            amountGovTokens,
```
Comment (0x4007) [inline packages/contracts/src/dollar/core/DollarMintExcess.sol]:
```suggestion
            amountGovernanceTokens,
```

File: packages/contracts/src/dollar/interfaces/IUbiquityDollarToken.sol [hunk 7]
```diff
@@ -3,9 +3,9 @@ pragma solidity ^0.8.3;
 
 import "./IERC20Ubiquity.sol";
 
-/// @title UAD stablecoin interface
-/// @author Ubiquity Algorithmic Dollar
-interface IUbiquityAlgorithmicDollar is IERC20Ubiquity {
+/// @title Ubiquity Dollar stablecoin interface
+/// @author Ubiquity Dollar
```
Comment (0x4007) [inline packages/contracts/src/dollar/interfaces/IUbiquityDollarToken.sol]:
```suggestion
/// @author Ubiquity DAO
```

File: packages/contracts/src/dollar/interfaces/IUbiquityGovernance.sol [hunk 8]
```diff
@@ -3,7 +3,7 @@ pragma solidity ^0.8.3;
 
 import "./IERC20Ubiquity.sol";
 
-/// @title uGOV governance token interface
-/// @author Ubiquity Algorithmic Dollar
+/// @title Governance token interface
+/// @author Ubiquity Dollar
```
Comment (0x4007) [inline packages/contracts/src/dollar/interfaces/IUbiquityGovernance.sol]:
```suggestion
/// @author Ubiquity DAO
```

File: packages/contracts/src/dollar/transfer-hooks/CurveDollarIncentive.sol [hunk 9]
```diff
@@ -1,21 +1,21 @@
...
-/// @author uAD Protocol
+import "../core/TWAPOracleDollar3pool.sol";
+import "../core/UbiquityDollarManager.sol";
+import "../core/UbiquityDollarToken.sol";
+import "../interfaces/IUbiquityGovernance.sol";
+import "../interfaces/IIncentive.sol";
+import "../libs/ABDKMathQuad.sol";
+
+/// @title Curve trading incentive contract
+/// @author Ubiquity Protocol
```
Comment (0x4007) [inline packages/contracts/src/dollar/transfer-hooks/CurveDollarIncentive.sol]:
```suggestion
/// @author Ubiquity DAO

```

I think I don't need to keep suggesting this change, you know to change this if you see it elsewhere in the code!

File: packages/contracts/test/dollar/ERC1155Ubiquity.t.sol [hunk 10]
```diff
@@ -2,13 +2,13 @@
...
-import {UbiquityAlgorithmicDollarManager} from
-    "../../src/dollar/UbiquityAlgorithmicDollarManager.sol";
+import {UbiquityDollarManager} from
+    "../../src/dollar/core/UbiquityDollarManager.sol";
 import "../helpers/LocalTestHelper.sol";
 
 contract ERC1155UbiquityTest is LocalTestHelper {
     address token_addr;
-    address uad_manager_addr;
+    address dollar_manager_addr;
```
Comment (0x4007) [inline packages/contracts/test/dollar/ERC1155Ubiquity.t.sol:11]: Seems unusual to me that the variables in some of these tests are snake case @0xcodercrane  why is that

File: packages/contracts/test/dollar/UbiquityChef.t.sol [hunk 11]
```diff
@@ -22,49 +22,49 @@ contract ZeroState is LiveTestHelper {
...
 contract ZeroStateTest is ZeroState {
     function testSetUGOVPerBlock(uint256 uGOVPerBlock) public {
-        vm.expectEmit(true, false, false, true, address(chefV2));
+        vm.expectEmit(true, false, false, true, address(ubiquityChef));
         emit UGOVPerBlockModified(uGOVPerBlock);
         vm.prank(admin);
-        chefV2.setUGOVPerBlock(uGOVPerBlock);
-        assertEq(chefV2.uGOVPerBlock(), uGOVPerBlock);
+        ubiquityChef.setUGOVPerBlock(uGOVPerBlock);
+        assertEq(ubiquityChef.uGOVPerBlock(), uGOVPerBlock);
```
Comment (0x4007) [inline packages/contracts/test/dollar/UbiquityChef.t.sol]: I expect that the method names should have been updated to the new naming convention.

File: packages/contracts/test/helpers/LiveTestHelper.sol [hunk 12]
```diff
@@ -1,62 +1,62 @@
...
-    TWAPOracle twapOracle;
-    MasterChefV2 chefV2;
-    UARForDollarsCalculator uarCalc;
-    CouponsForDollarsCalculator couponCalc;
-    DollarMintingCalculator dollarMintCalc;
-    MockDebtCoupon debtCoupon;
-    DebtCouponManager debtCouponMgr;
-    UbiquityAutoRedeem uAR;
-    ExcessDollarsDistributor excessDollarsDistributor;
+    Bonding stakingV1;
```
Comment (0x4007) [inline packages/contracts/test/helpers/LiveTestHelper.sol:36]: Do we need versioning? I expect that we don't.

File: packages/contracts/test/helpers/LiveTestHelper.sol [hunk 13]
```diff
@@ -100,50 +100,50 @@ contract LiveTestHelper is Test {
...
+        manager.setStakingContractAddress(address(stakingV1));
 
-        bondingShareV1.setApprovalForAll(address(bondingV1), true);
+        stakingShareV1.setApprovalForAll(address(stakingV1), true);
 
         manager.setTreasuryAddress(treasury);
 
-        deal(address(uGov), thirdAccount, 100000e18);
-        deal(address(uAD), thirdAccount, 1000000e18);
+        deal(address(govToken), thirdAccount, 100000e18);
```
Comment (0x4007) [inline packages/contracts/test/helpers/LiveTestHelper.sol]:
```suggestion
        deal(address(governanceToken), thirdAccount, 100000e18);
```

File: packages/contracts/test/helpers/LiveTestHelper.sol [hunk 14]
```diff
@@ -159,26 +159,26 @@ contract LiveTestHelper is Test {
...
         ];
 
         for (uint256 i = 0; i < mintings.length; ++i) {
-            deal(address(uAD), mintings[i], 10000e18);
+            deal(address(dollarToken), mintings[i], 10000e18);
         }
 
-        manager.grantRole(manager.UBQ_MINTER_ROLE(), address(bondingV1));
-        manager.grantRole(manager.UBQ_BURNER_ROLE(), address(bondingV1));
+        manager.grantRole(manager.GOV_TOKEN_MINTER_ROLE(), address(stakingV1));
```
Comment (0x4007) [inline packages/contracts/test/helpers/LiveTestHelper.sol]:
```suggestion
        manager.grantRole(manager.GOVERNANCE_TOKEN_MINTER_ROLE(), address(stakingV1));
```

I think we should try and minimize the shorthand use of "GOV"

File: packages/contracts/test/helpers/LocalTestHelper.sol [hunk 15]
```diff
@@ -38,51 +38,51 @@ abstract contract LocalTestHelper is Test {
...
         // deploy governance token
-        UbiquityGovernance _uGov = new UbiquityGovernance(address(_manager));
-        _manager.setGovernanceTokenAddress(address(_uGov));
-
-        // deploy couponsForDollarCalculator
-        MockCouponsForDollarsCalculator couponsForDollarsCalculator =
-            new MockCouponsForDollarsCalculator();
-        _manager.setCouponCalculatorAddress(
-            address(couponsForDollarsCalculator)
+        UbiquityGovernanceToken _govToken = new UbiquityGovernanceToken(address(_manager));
```
Comment (0x4007) [inline packages/contracts/test/helpers/LocalTestHelper.sol]:
```suggestion
        UbiquityGovernanceToken _governanceToken = new UbiquityGovernanceToken(address(_manager));
```

Refer to my previous comment

**Review by 0x4007 — CHANGES_REQUESTED** [review]
I reviewed on my phone so I might have missed a few details but overall this is a very pleasant PR to see. The codebase seems to be a lot more understandable after these renames. Thank you for your attention to detail, per usual.

File: packages/contracts/src/dollar/core/DollarMintCalculator.sol (hunk 3 above)
Comment (ryzhak) [reply packages/contracts/src/dollar/core/DollarMintCalculator.sol:11]: I can't understand what you mean by "same tense of the verbs". Need a concrete example of what's wrong.

Comment (0x4007) [reply packages/contracts/src/dollar/core/DollarMintCalculator.sol:11]: This was my assumption for why you renamed from `DollarMintingCalculator` to `DollarMintCalculator` it's not an issue.

Comment (ryzhak) [reply packages/contracts/src/dollar/core/DollarMintCalculator.sol:11]: this renaming was described in the task

`DollarMintingCalculator.sol ⮕ DollarMintCalculator.sol `

**Comment by ryzhak** [comment]
@pavlovcik @0xcodercrane 

guys pls check

File: packages/contracts/test/helpers/LiveTestHelper.sol (hunk 12 above)
Comment (ryzhak) [reply packages/contracts/test/helpers/LiveTestHelper.sol:36]: this is intentional

how it was:
```
Bonding bondingV1;
BondingV2 bondingV2;
```

how it is now:
```
Bonding stakingV1;
Staking staking;
```

so this is for new developers to understand that now we use `Staking staking`, but earlier we used `Bonding stakingV1;` and that "bonding" is the same thing as "staking"

this is complicated nevetherless)

**Review by 0x4007 — APPROVED** [review]
Looks awesome! I would feel most comfortable with @0xcodercrane approving the Solidity changes but I would be ok with releasing the bounty before we merge this too. 

I have a feeling that all the code changes could really mess with other open PRs so I want to consult with the team before merging.

**Review by cuddle-dev — APPROVED** [review]
Looks good to me

**Comment by hashedMae** [comment]
Unless we're planning on redeploying and reminting the UBQ token to all current holders then changes to `UbiquityGovernanceToken.sol` should be reverted, especially the minter roles with `UbiquityDollarManager`.
`uGOV` was never a good variable as there is no token with that symbol and the contracts should still be updated to reflect that.

**Comment by 0x4007** [comment]
> Unless we're planning on redeploying and reminting the UBQ token to all current holders

We could do a "spoon" and distribute. With only 105 holders, doing an airdrop is definitely in the cards.

**Pull request merged**